manifests/server.pp

```diff
@@ -158,18 +159,43 @@
           }
         }
       }
+    }
 
-      # Wrap the replset class
-      class { 'mongodb::replset':
-        sets => $replset_config_REAL
-      }
-      Anchor['mongodb::server::end'] -> Class['mongodb::replset']
+    # Wrap the replset class
+    class { 'mongodb::replset':
+      sets => $replset_config_REAL
+    }
+    Anchor['mongodb::server::end'] -> Class['mongodb::replset']
+
+    # Make sure that the ordering is correct
+    if $create_admin {
+      Class['mongodb::replset'] -> Mongodb::Db['admin']
```

Comment:
Similarly here, if `$replset` is false and `Class['mongodb::replset']` is then not declared, this dependency would be invalid so should be moved into a `if $replset and $create_admin` conditional